SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialComponentTest/SpatialComponentTest.cpp

```diff
@@ -0,0 +1,231 @@
+	if (LocalWorkerDefinition.Type == ESpatialFunctionalTestWorkerType::Server)
+	{
+		// Allow it to continue working in Native / Single worker setups.
+		if (GetNumberOfServerWorkers() > 1)
+		{
+			if (LocalWorkerDefinition.Id == ExpectedServerId)
+			{
+				RequireTrue(VerifyTestActorComponents(Actor, 2), "Server auth - OnAuthorityGained component and OnActorReady component");
+				FinishStep();
+			}
+			else if (Actor->bNetStartup)
+			{
+				RequireTrue(VerifyTestActorComponents(Actor, 1),
+							"Non-auth servers - Level actors receive OnActorReady only OnAuthorityGained");
+				FinishStep();
+			}
+			else
+			{
+				RequireTrue(VerifyTestActorComponents(Actor, 0),
+							"Non-auth servers - Dynamic actors do not receive OnActorReady or OnAuthorityGained ");
+				FinishStep();
+			}
+		}
+		else // Support for Native / Single Worker.
+		{
+			RequireTrue(VerifyTestActorComponents(Actor, 2),
+						"Native / Single Worker - OnActorReady component and OnAuthorityGained component");
+			FinishStep();
+		}
+	}
+	else // Clients
+	{
+		if (LocalWorkerDefinition.Id == 1)
+		{
+			RequireTrue(VerifyTestActorComponents(Actor, ExpectedClient1ComponentCount), "Client 1");
+			FinishStep();
+		}
+		else if (LocalWorkerDefinition.Id == 2)
+		{
+			RequireTrue(VerifyTestActorComponents(Actor, ExpectedClient2ComponentCount), "Client 2");
+			FinishStep();
+		}
+	}
+}
+
+// Checks the number of components on the servers and clients when an actor migrates
+void ASpatialComponentTest::CheckComponentsCrossServer(ASpatialComponentTestActor* Actor, int StartServerId, int EndServerId)
+{
+	const FWorkerDefinition& LocalWorkerDefinition = GetLocalFlowController()->WorkerDefinition;
+	if (LocalWorkerDefinition.Type == ESpatialFunctionalTestWorkerType::Server)
+	{
+		// Allow it to continue working in Native / Single worker setups.
+		if (GetNumberOfServerWorkers() > 1)
+		{
+			if (LocalWorkerDefinition.Id == StartServerId)
+			{
+				RequireTrue(VerifyTestActorComponents(Actor, 3),
+							"Spawning server - OnActorReady component, OnAuthorityGained component and OnAuthorityLost component");
+				FinishStep();
+			}
+			else if (LocalWorkerDefinition.Id == EndServerId)
+			{
+				RequireTrue(VerifyTestActorComponents(Actor, 1), "Migrated server - OnAuthorityGained component");
+				FinishStep();
+			}
+		}
+		else // Support for Native / Single Worker.
+		{
+			RequireTrue(VerifyTestActorComponents(Actor, 2),
+						"Native / Single Worker - OnActorReady component and OnAuthorityGained component");
+			FinishStep();
+		}
+	}
+	else // Clients
+	{
+		RequireTrue(VerifyTestActorComponents(Actor, 0), "Clients");
+		FinishStep();
+	}
+}
+
+bool ASpatialComponentTest::VerifyTestActorComponents(ASpatialComponentTestActor* Actor, int ExpectedTestComponentCount)
+{
+	TArray<UActorComponent*> FoundComponents = Actor->GetComponentsByClass(USpatialComponentTestDummyComponent::StaticClass());
```

Comment:
Might be good to also validate that BeginPlay has been called by this point, to ensure the 0 test cases are correct.

Reply:
Good point. Done.